### PR TITLE
Allow forks to org if you can create repos

### PR DESCRIPTION
**Pull request by qwerty287**
Makes it possible to fork a repo to an org if you can create repos in this org. Currently, that's only available to owners.

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

File: routers/web/repo/pull.go [hunk 1]
```diff
@@ -217,9 +217,9 @@ func ForkPost(ctx *context.Context) {
 
 	// Check ownership of organization.
 	if ctxUser.IsOrganization() {
-		isOwner, err := models.OrgFromUser(ctxUser).IsOwnedBy(ctx.User.ID)
+		isOwner, err := models.OrgFromUser(ctxUser).CanCreateOrgRepo(ctx.User.ID)
```
Comment (Gusted) [inline routers/web/repo/pull.go]: maybe change the the name of this variable. `isAllowedToFork`

Comment (lunny) [reply routers/web/repo/pull.go]: The variable name `isOwner` should also be changed.

Comment (Gusted) [reply routers/web/repo/pull.go]: And the comment on Line 218 ;)

File: routers/web/repo/pull.go [hunk 2]
```diff
@@ -217,9 +217,9 @@ func ForkPost(ctx *context.Context) {
 
 	// Check ownership of organization.
```
Comment (Gusted) [inline routers/web/repo/pull.go]:
```suggestion
	// Check if user is allowed to create repo's on the organization.
```

File: routers/web/repo/pull.go (hunk 1 above)
Comment (Gusted) [inline routers/web/repo/pull.go]:
```suggestion
		isAllowedToFork, err := models.OrgFromUser(ctxUser).CanCreateOrgRepo(ctx.User.ID)
```

File: routers/web/repo/pull.go [hunk 3]
```diff
@@ -217,9 +217,9 @@ func ForkPost(ctx *context.Context) {
...
 	if ctxUser.IsOrganization() {
-		isOwner, err := models.OrgFromUser(ctxUser).IsOwnedBy(ctx.User.ID)
+		isOwner, err := models.OrgFromUser(ctxUser).CanCreateOrgRepo(ctx.User.ID)
 		if err != nil {
-			ctx.ServerError("IsOwnedBy", err)
+			ctx.ServerError("CanCreateOrgRepo", err)
 			return
 		} else if !isOwner {
```
Comment (Gusted) [inline routers/web/repo/pull.go]:
```suggestion
		} else if !isAllowedToFork {
```

**Review by Gusted — COMMENTED** [review]
Commit it, and good to go.

**Comment by qwerty287** [comment]
@Gusted Sorry, I didn't had time to this until now...

**Review by Gusted — APPROVED** [review]

**Comment by Gusted** [comment]
> @Gusted Sorry, I didn't have time to do this until now...

No worries.

**Review by lunny — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

File: routers/web/repo/pull.go [hunk 4]
```diff
@@ -113,9 +113,9 @@ func getForkRepository(ctx *context.Context) *models.Repository {
 
 	ctx.Data["ForkRepo"] = forkRepo
 
-	ownedOrgs, err := models.GetOwnedOrgsByUserID(ctx.User.ID)
+	ownedOrgs, err := models.GetOrgsCanCreateRepoByUserID(ctx.User.ID)
```
Comment (lunny) [inline routers/web/repo/pull.go:116]: There is an performance regression for all pages with fork button. This should be fixed in v1.16.

**Pull request merged**